code/src/app/unitenew/uniteServices/routingService.ts

```diff
@@ -170,4 +174,37 @@ export class UniteRouting{
         return menusToReturn;
 
     }
+
+    getAllWidgets()
+    {
+        this._widget.getWidgets().subscribe(data =>{
+            this.mapWidgetsWithPages(data);
+        });
+    }
+
+    mapWidgetsWithPages(widgets)
+    {
+        let i = 0;
```

Comment:
@ttplsudhir : No need of explicit incremental variable for indexing. `forEach` does provides an index parameter. Check out this - https://www.w3schools.com/jsref/jsref_forEach.asp

Reply:
@sushan05 Yes will update the code accordingly